frame/support/src/weights.rs

```diff
@@ -338,7 +381,10 @@ impl<T> WithPostDispatchInfo for T where
 {
 	fn with_weight(self, actual_weight: Weight) -> DispatchErrorWithPostInfo {
 		DispatchErrorWithPostInfo {
-			post_info: PostDispatchInfo { actual_weight: Some(actual_weight) },
+			post_info: PostDispatchInfo {
+				actual_weight: Some(actual_weight),
+				pays_fee: Default::default(),
```

Comment:
Question: This will always return `pays_fee` as `Pays::Yes` which is fine even if the dispatchable was originally intended to pay no fees because `pays_fee` takes into account both pre and post?

Reply:
if info.pays_fee is no and postinfo.pays_fee is yes then no fee are paid. post info can only reduce fees and/or weight

Reply:
Yeah, exactly what @thiolliere said. Look at this: https://github.com/paritytech/substrate/pull/6749/files#diff-c20449a380ebb6d2f477378f22ba3b75R291

Reply:
@thiolliere @shawntabrizi  Yes, I was confirming my understanding. Thanks.